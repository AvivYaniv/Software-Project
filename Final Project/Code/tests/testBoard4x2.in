edit tests/b4_2.txt
set 7 5 1
reset
num_solutions
hint 1 1
validate
set 6 2 3
save tests/observed/b4_2-saved.txt
solve tests/observed/b4_2-saved.txt
print_board
set 7 1 2
validate 
num_solutions
reset
autofill
undo
redo
validate
num_solutions
set 1 2 4
hint 1 1
hint 2 1
hint 1 3
autofill
hint 1 1
set 4 3 3
undo
set 4 3 1
undo
save tests/observed/b4_2-saved2.txt
solve tests/observed/b4_2-saved2.txt
set 1 1 7
set 4 1 7
undo
undo
validate
autofill