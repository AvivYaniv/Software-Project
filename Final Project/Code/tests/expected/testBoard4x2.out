Sudoku
------
Enter your command:
-------------------------------------
|        |  7     |      8 |      4 |
|  2   7 |  8     |  4     |  6   1 |
|      3 |  2     |      1 |  8   7 |
|  4     |  1   6 |      5 |      3 |
-------------------------------------
|      5 |  4   8 |      2 |      6 |
|      4 |  5   7 |      6 |  3   2 |
|      2 |      1 |  3   7 |  4   5 |
|        |      2 |      4 |  1   8 |
-------------------------------------
Enter your command:
-------------------------------------
|        |  7     |      8 |      4 |
|  2   7 |  8     |  4     |  6   1 |
|      3 |  2     |      1 |  8   7 |
|  4     |  1   6 |      5 |      3 |
-------------------------------------
|      5 |  4   8 |      2 |  1*  6 |
|      4 |  5   7 |      6 |  3   2 |
|      2 |      1 |  3   7 |  4   5 |
|        |      2 |      4 |  1*  8 |
-------------------------------------
Enter your command:
Board reset
Enter your command:
Number of solutions: 1
This is a good board!
Enter your command:
ERROR: invalid command
Enter your command:
Validation passed: board is solvable
Enter your command:
-------------------------------------
|        |  7     |      8 |      4 |
|  2   7 |  8     |  4   3 |  6   1 |
|      3 |  2     |      1 |  8   7 |
|  4     |  1   6 |      5 |      3 |
-------------------------------------
|      5 |  4   8 |      2 |      6 |
|      4 |  5   7 |      6 |  3   2 |
|      2 |      1 |  3   7 |  4   5 |
|        |      2 |      4 |  1   8 |
-------------------------------------
Enter your command:
Saved to: tests/observed/b4_2-saved.txt
Enter your command:
-------------------------------------
|        |  7.    |      8.|      4.|
|  2.  7.|  8.    |  4.  3.|  6.  1.|
|      3.|  2.    |      1.|  8.  7.|
|  4.    |  1.  6.|      5.|      3.|
-------------------------------------
|      5.|  4.  8.|      2.|      6.|
|      4.|  5.  7.|      6.|  3.  2.|
|      2.|      1.|  3.  7.|  4.  5.|
|        |      2.|      4.|  1.  8.|
-------------------------------------
Enter your command:
-------------------------------------
|        |  7.    |      8.|      4.|
|  2.  7.|  8.    |  4.  3.|  6.  1.|
|      3.|  2.    |      1.|  8.  7.|
|  4.    |  1.  6.|      5.|      3.|
-------------------------------------
|      5.|  4.  8.|      2.|      6.|
|      4.|  5.  7.|      6.|  3.  2.|
|      2.|      1.|  3.  7.|  4.  5.|
|        |      2.|      4.|  1.  8.|
-------------------------------------
Enter your command:
-------------------------------------
|        |  7.    |      8.|  2   4.|
|  2.  7.|  8.    |  4.  3.|  6.  1.|
|      3.|  2.    |      1.|  8.  7.|
|  4.    |  1.  6.|      5.|      3.|
-------------------------------------
|      5.|  4.  8.|      2.|      6.|
|      4.|  5.  7.|      6.|  3.  2.|
|      2.|      1.|  3.  7.|  4.  5.|
|        |      2.|      4.|  1.  8.|
-------------------------------------
Enter your command:
Validation failed: board is unsolvable
Enter your command:
Number of solutions: 0
Enter your command:
Board reset
Enter your command:
Cell <4,2> set to 5
Cell <5,3> set to 6
Cell <2,4> set to 8
Cell <7,4> set to 2
Cell <5,5> set to 1
Cell <7,5> set to 7
Cell <3,7> set to 6
Cell <2,8> set to 6
Cell <5,8> set to 5
-------------------------------------
|        |  7.    |      8.|      4.|
|  2.  7.|  8.  5 |  4.  3.|  6.  1.|
|      3.|  2.    |  6   1.|  8.  7.|
|  4.  8 |  1.  6.|      5.|  2   3.|
-------------------------------------
|      5.|  4.  8.|  1   2.|  7   6.|
|      4.|  5.  7.|      6.|  3.  2.|
|      2.|  6   1.|  3.  7.|  4.  5.|
|      6 |      2.|  5   4.|  1.  8.|
-------------------------------------
Enter your command:
-------------------------------------
|        |  7.    |      8.|      4.|
|  2.  7.|  8.    |  4.  3.|  6.  1.|
|      3.|  2.    |      1.|  8.  7.|
|  4.    |  1.  6.|      5.|      3.|
-------------------------------------
|      5.|  4.  8.|      2.|      6.|
|      4.|  5.  7.|      6.|  3.  2.|
|      2.|      1.|  3.  7.|  4.  5.|
|        |      2.|      4.|  1.  8.|
-------------------------------------
Undo 4,2: from 5 to _
Undo 5,3: from 6 to _
Undo 2,4: from 8 to _
Undo 7,4: from 2 to _
Undo 5,5: from 1 to _
Undo 7,5: from 7 to _
Undo 3,7: from 6 to _
Undo 2,8: from 6 to _
Undo 5,8: from 5 to _
Enter your command:
-------------------------------------
|        |  7.    |      8.|      4.|
|  2.  7.|  8.  5 |  4.  3.|  6.  1.|
|      3.|  2.    |  6   1.|  8.  7.|
|  4.  8 |  1.  6.|      5.|  2   3.|
-------------------------------------
|      5.|  4.  8.|  1   2.|  7   6.|
|      4.|  5.  7.|      6.|  3.  2.|
|      2.|  6   1.|  3.  7.|  4.  5.|
|      6 |      2.|  5   4.|  1.  8.|
-------------------------------------
Redo 4,2: from _ to 5
Redo 5,3: from _ to 6
Redo 2,4: from _ to 8
Redo 7,4: from _ to 2
Redo 5,5: from _ to 1
Redo 7,5: from _ to 7
Redo 3,7: from _ to 6
Redo 2,8: from _ to 6
Redo 5,8: from _ to 5
Enter your command:
Validation passed: board is solvable
Enter your command:
Number of solutions: 1
This is a good board!
Enter your command:
Error: cell is fixed
Enter your command:
Hint: set cell to 6
Enter your command:
Hint: set cell to 1
Enter your command:
Hint: set cell to 5
Enter your command:
Cell <2,1> set to 1
Cell <4,1> set to 3
Cell <5,1> set to 2
Cell <7,1> set to 5
Cell <1,3> set to 5
Cell <4,3> set to 4
Cell <5,4> set to 7
Cell <1,5> set to 3
Cell <5,6> set to 8
Cell <1,7> set to 8
Cell <3,8> set to 3
-------------------------------------
|      1 |  7.  3 |  2   8.|  5   4.|
|  2.  7.|  8.  5 |  4.  3.|  6.  1.|
|  5   3.|  2.  4 |  6   1.|  8.  7.|
|  4.  8 |  1.  6.|  7   5.|  2   3.|
-------------------------------------
|  3   5.|  4.  8.|  1   2.|  7   6.|
|      4.|  5.  7.|  8   6.|  3.  2.|
|  8   2.|  6   1.|  3.  7.|  4.  5.|
|      6 |  3   2.|  5   4.|  1.  8.|
-------------------------------------
Enter your command:
Hint: set cell to 6
Enter your command:
-------------------------------------
|      1 |  7.  3*|  2   8.|  5   4.|
|  2.  7.|  8.  5 |  4.  3.|  6.  1.|
|  5   3.|  2.  3*|  6   1.|  8.  7.|
|  4.  8 |  1.  6.|  7   5.|  2   3.|
-------------------------------------
|  3   5.|  4.  8.|  1   2.|  7   6.|
|      4.|  5.  7.|  8   6.|  3.  2.|
|  8   2.|  6   1.|  3.  7.|  4.  5.|
|      6 |  3   2.|  5   4.|  1.  8.|
-------------------------------------
Enter your command:
-------------------------------------
|      1 |  7.  3 |  2   8.|  5   4.|
|  2.  7.|  8.  5 |  4.  3.|  6.  1.|
|  5   3.|  2.  4 |  6   1.|  8.  7.|
|  4.  8 |  1.  6.|  7   5.|  2   3.|
-------------------------------------
|  3   5.|  4.  8.|  1   2.|  7   6.|
|      4.|  5.  7.|  8   6.|  3.  2.|
|  8   2.|  6   1.|  3.  7.|  4.  5.|
|      6 |  3   2.|  5   4.|  1.  8.|
-------------------------------------
Undo 4,3: from 3 to 4
Enter your command:
-------------------------------------
|      1 |  7.  3 |  2   8.|  5   4.|
|  2.  7.|  8.  5 |  4.  3.|  6.  1.|
|  5   3.|  2.  1*|  6   1.|  8.  7.|
|  4.  8 |  1.  6.|  7   5.|  2   3.|
-------------------------------------
|  3   5.|  4.  8.|  1   2.|  7   6.|
|      4.|  5.  7.|  8   6.|  3.  2.|
|  8   2.|  6   1.|  3.  7.|  4.  5.|
|      6 |  3   2.|  5   4.|  1.  8.|
-------------------------------------
Enter your command:
-------------------------------------
|      1 |  7.  3 |  2   8.|  5   4.|
|  2.  7.|  8.  5 |  4.  3.|  6.  1.|
|  5   3.|  2.  4 |  6   1.|  8.  7.|
|  4.  8 |  1.  6.|  7   5.|  2   3.|
-------------------------------------
|  3   5.|  4.  8.|  1   2.|  7   6.|
|      4.|  5.  7.|  8   6.|  3.  2.|
|  8   2.|  6   1.|  3.  7.|  4.  5.|
|      6 |  3   2.|  5   4.|  1.  8.|
-------------------------------------
Undo 4,3: from 1 to 4
Enter your command:
Saved to: tests/observed/b4_2-saved2.txt
Enter your command:
-------------------------------------
|      1 |  7.  3 |  2   8.|  5   4.|
|  2.  7.|  8.  5 |  4.  3.|  6.  1.|
|  5   3.|  2.  4 |  6   1.|  8.  7.|
|  4.  8 |  1.  6.|  7   5.|  2   3.|
-------------------------------------
|  3   5.|  4.  8.|  1   2.|  7   6.|
|      4.|  5.  7.|  8   6.|  3.  2.|
|  8   2.|  6   1.|  3.  7.|  4.  5.|
|      6 |  3   2.|  5   4.|  1.  8.|
-------------------------------------
Enter your command:
-------------------------------------
|  7*  1 |  7.  3 |  2   8.|  5   4.|
|  2.  7.|  8.  5 |  4.  3.|  6.  1.|
|  5   3.|  2.  4 |  6   1.|  8.  7.|
|  4.  8 |  1.  6.|  7   5.|  2   3.|
-------------------------------------
|  3   5.|  4.  8.|  1   2.|  7   6.|
|      4.|  5.  7.|  8   6.|  3.  2.|
|  8   2.|  6   1.|  3.  7.|  4.  5.|
|      6 |  3   2.|  5   4.|  1.  8.|
-------------------------------------
Enter your command:
-------------------------------------
|  7*  1 |  7.  7*|  2   8.|  5   4.|
|  2.  7.|  8.  5 |  4.  3.|  6.  1.|
|  5   3.|  2.  4 |  6   1.|  8.  7.|
|  4.  8 |  1.  6.|  7   5.|  2   3.|
-------------------------------------
|  3   5.|  4.  8.|  1   2.|  7   6.|
|      4.|  5.  7.|  8   6.|  3.  2.|
|  8   2.|  6   1.|  3.  7.|  4.  5.|
|      6 |  3   2.|  5   4.|  1.  8.|
-------------------------------------
Enter your command:
-------------------------------------
|  7*  1 |  7.  3 |  2   8.|  5   4.|
|  2.  7.|  8.  5 |  4.  3.|  6.  1.|
|  5   3.|  2.  4 |  6   1.|  8.  7.|
|  4.  8 |  1.  6.|  7   5.|  2   3.|
-------------------------------------
|  3   5.|  4.  8.|  1   2.|  7   6.|
|      4.|  5.  7.|  8   6.|  3.  2.|
|  8   2.|  6   1.|  3.  7.|  4.  5.|
|      6 |  3   2.|  5   4.|  1.  8.|
-------------------------------------
Undo 4,1: from 7 to 3
Enter your command:
-------------------------------------
|      1 |  7.  3 |  2   8.|  5   4.|
|  2.  7.|  8.  5 |  4.  3.|  6.  1.|
|  5   3.|  2.  4 |  6   1.|  8.  7.|
|  4.  8 |  1.  6.|  7   5.|  2   3.|
-------------------------------------
|  3   5.|  4.  8.|  1   2.|  7   6.|
|      4.|  5.  7.|  8   6.|  3.  2.|
|  8   2.|  6   1.|  3.  7.|  4.  5.|
|      6 |  3   2.|  5   4.|  1.  8.|
-------------------------------------
Undo 1,1: from 7 to _
Enter your command:
Validation passed: board is solvable
Enter your command:
Cell <1,1> set to 6
Cell <1,6> set to 1
Cell <1,8> set to 7
-------------------------------------
|  6   1 |  7.  3 |  2   8.|  5   4.|
|  2.  7.|  8.  5 |  4.  3.|  6.  1.|
|  5   3.|  2.  4 |  6   1.|  8.  7.|
|  4.  8 |  1.  6.|  7   5.|  2   3.|
-------------------------------------
|  3   5.|  4.  8.|  1   2.|  7   6.|
|  1   4.|  5.  7.|  8   6.|  3.  2.|
|  8   2.|  6   1.|  3.  7.|  4.  5.|
|  7   6 |  3   2.|  5   4.|  1.  8.|
-------------------------------------
Puzzle solved successfully
Enter your command:
Exiting...
